content/hacking-atom/sections/creating-a-grammar.md

```diff
@@ -0,0 +1,186 @@
+---
+title: Creating a Grammar
+---
+### Creating a Grammar
+
+Grammars power Atom's syntax highlighting. Each grammar provides a set of _regex_ patterns which are assigned _scopes_. These scopes are then turned into the CSS classes that you can target in syntax themes!
+
+#### Getting Started
+
+Grammars depend heavily on regexes, and you should be comfortable with interpreting and writing regexes before continuing. Note that Atom uses the Oniguruma engine, which is very similar to the PCRE or Perl regex engines. Here are some resources to help you out:
+
+* http://www.regular-expressions.info/tutorial.html provides a comprehensive regex tutorial
+* http://www.rexegg.com/regex-quickstart.html contains a cheat sheet for various regex expressions
+* https://regex101.com/ or https://regexr.com/ allows live prototyping
+
+Grammar files are written in the [CSON](https://github.com/bevry/cson#what-is-cson) or [JSON](http://www.json.org/) format. Whichever one you decide to use is up to you, but this tutorial will be written in CSON.
```

Comment:
I'm going to be very pedantic here, but I think that this would read better as, "Grammar files are written in either CSON or JSON format."